miniconf/tests/packed.rs

```diff
@@ -0,0 +1,71 @@
+#![cfg(feature = "json-core")]
+
+use miniconf::{Error, Packed, Tree, TreeKey, TreeSerialize};
+
+#[derive(Tree, Default)]
+struct Settings {
+    a: f32,
+    #[tree(depth = 1)]
+    b: [f32; 2],
+}
+
+#[test]
+fn packed() {
+    let mut p = String::new();
+
+    assert_eq!(
+        Settings::path(Packed::default(), &mut p, "/"),
+        Err(Error::TooShort(0))
+    );
+    p.clear();
+
+    for q in Settings::iter_paths::<String>("/") {
+        let q = q.unwrap();
+        let (a, _d) = Settings::packed(q.split("/").skip(1)).unwrap();
+        Settings::path(a, &mut p, "/").unwrap();
+        assert_eq!(p, q);
+        p.clear();
+    }
+    println!(
+        "{:?}",
+        Settings::iter_packed()
+            .map(Result::unwrap)
+            .collect::<Vec<_>>()
+    );
+
+    assert_eq!(
+        Settings::path(Packed::new(0b01 << 29).unwrap(), &mut p, "/"),
```

Comment:
Not sure what this is testing? Is this checking the returned depth?

Reply:
Yes. It's asserting that `path()` succeeds with the correct depth.